File: groups/group1/Euclidean.java
import java.util.Scanner;

public class Euclidean {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        String firstText = scanner.hasNextLine() ? scanner.nextLine() : "";
        String secondText = scanner.hasNextLine() ? scanner.nextLine() : "";

        if (!hasValidInput(firstText, secondText)) {
            return;
        }

        int first = Integer.parseInt(firstText.trim());
        int second = Integer.parseInt(secondText.trim());
        int a = first;
        int b = second;

        while (b != 0) {
            int remainder = a % b;
            a = b;
            b = remainder;
        }

        System.out.println("First number: " + first);
        System.out.println("Second number: " + second);
        System.out.println("GCD: " + a);
    }

    static boolean hasValidInput(String firstText, String secondText) {
        if (!isInteger(firstText)) {
            System.out.println("Invalid input. Enter the first positive integer.");
            return false;
        }

        if (!isInteger(secondText)) {
            System.out.println("Invalid input. Enter the second positive integer.");
            return false;
        }

        int first = Integer.parseInt(firstText.trim());
        int second = Integer.parseInt(secondText.trim());

        if (first <= 0 || second <= 0) {
            System.out.println("Invalid input. Both numbers must be positive integers.");
            return false;
        }

        return true;
    }

    static boolean isInteger(String value) {
        try {
            Integer.parseInt(value.trim());
            return true;
        } catch (NumberFormatException error) {
            return false;
        }
    }
}

File: groups/group1/Euclidean.js
function calculateEuclidean(m, n) {
  const origM = m;
  const origN = n;
  const lines = ["", "SOLUTION:"];
  let gcd;

  while (true) {
    const q = Math.floor(m / n);
    const r = m % n;

    if (r === 0) {
      lines.push(`${m} = ${n} (${q})`);
      gcd = n;
      break;
    }

    lines.push(`${m} = ${n} (${q}) + ${r}`);
    m = n;
    n = r;
  }

  const lcm = (origM * origN) / gcd;

  lines.push("");
  lines.push(`The integers are ${origM} and ${origN}`);
  lines.push(`The greatest common divisor of ${origM} and ${origN} is ${gcd}`);
  lines.push(`The least common multiple of ${origM} and ${origN} is ${lcm}`);

  return lines.join("\n");
}

function parsePositiveInt(rawValue) {
  const trimmed = String(rawValue ?? "").trim();

  if (!/^-?\d+$/.test(trimmed)) {
    throw new Error("Invalid input. Please enter an integer.");
  }

  const value = Number.parseInt(trimmed, 10);

  if (value <= 0) {
    throw new Error("Invalid input. Please enter a positive integer.");
  }

  return value;
}

function main(inputs) {
  const firstInput = parsePositiveInt(inputs[0]);
  const secondInput = parsePositiveInt(inputs[1]);

  const m = Math.max(firstInput, secondInput);
  const n = Math.min(firstInput, secondInput);

  return calculateEuclidean(m, n);
}
